sqlmesh/core/notification_target.py

```diff
@@ -396,12 +398,13 @@ def is_configured(self) -> bool:
 
 NotificationTarget = Annotated[
     t.Union[
+        BaseNotificationTarget,
         BasicSMTPNotificationTarget,
         ConsoleNotificationTarget,
         SlackApiNotificationTarget,
         SlackWebhookNotificationTarget,
     ],
-    Field(discriminator="type_"),
```

Comment:
[Pydantic Docs on Union/discriminators](https://docs.pydantic.dev/latest/concepts/unions/#union-validation-errors)

Changing from discriminator to union_mode "left_to_right" allows us to add BaseNotificationTarget and remove the error: `TypeError: Value 'smtp' for discriminator 'type_' mapped to multiple choices`